contracts/0.4.24/oracle/LidoOracle.sol

```diff
@@ -214,6 +263,34 @@ contract LidoOracle is ILidoOracle, AragonApp {
         _tryPush(_epochId);
     }
 
+    /**
+     * @notice To make oracles less dangerous,
+     *  we can bound rewards report by 0.1% increase in stake and 15% decrease in stake,
+     *  with both values configurable by DAO voting in case of extremely unusual circumstances.
+     *  daily_reward_rate_PPM = 1e6 * reward / totalPooledEther / days
+     **/
+    function reportSanityChecks(Report report) private {
+        (uint256 lastTotalPooledEther,
+         uint256 curTotalPooledEther,
+         uint256 lastValidators,
+         uint256 timeElapsed) = getLastCompletedReport();
```

Comment:
Well, it's not so good to call `getLastCompletedReport` here, because this function invokes `Lido` contract and we end up with 2 Lido calls per frame (frame=day here): one here and one in line#521. And Lido call is gas-expensive because of Aragon proxy (`AppProxyUpgradeable`). I believe we can save 1 call by doing this: 
- call `uint256 getTotalPooledEther = ISTETH(lido).totalSupply());` as in line#521,
- compare this `getTotalPooledEther` with its previous saved copy `LAST_COMPLETED_REPORT_TOTAL_SUPPLY.getStorageUint256()`,
- if the check passed, save the new value to the copy, otherwise revert.